### ISSUE TITLE
Try Semantic Labels

### ISSUE THREAD

**Issue by volodya-lombrozo** [volodya-lombrozo/jsmith#92]
As I have already described in the [Sematic Labels](https://github.com/volodya-lombrozo/jsmith?tab=readme-ov-file#semantic-labels) section, we might try to introduce some labels to ANTLR grammar:

```antlr
grammar Arithmetic;

prog: stat+ ;

stat: expr NEWLINE
    | ID '=' expr NEWLINE #ID_VariableDeclaration
    | NEWLINE
    ;

expr: expr ('*' | '/' ) expr
    | expr ('+' | '-' ) expr
    | INT
    | ID #ID_VariableUsage
    | '(' expr ')'
    ;

ID  : [a-zA-Z]+ ;
INT : [0-9]+ ;
NEWLINE: '\r'? '\n' ;
WS  : [ \t]+ -> skip ;
```

Pay attention to the `#ID_VariableDeclaration` and `#ID_VariableUsage` labels usage. Both of them are applied to `ID` lexer rule.
We should try if it's possible to guide program generation by using such labels.

**Comment by 0pdd** [comment]
@volodya-lombrozo 4 puzzles [#100](https://github.com/volodya-lombrozo/jsmith/issues/100), [#97](https://github.com/volodya-lombrozo/jsmith/issues/97), [#98](https://github.com/volodya-lombrozo/jsmith/issues/98), [#99](https://github.com/volodya-lombrozo/jsmith/issues/99) are still not solved.

**Comment by 0pdd** [comment]
@volodya-lombrozo 3 puzzles [#100](https://github.com/volodya-lombrozo/jsmith/issues/100), [#97](https://github.com/volodya-lombrozo/jsmith/issues/97), [#99](https://github.com/volodya-lombrozo/jsmith/issues/99) are still not solved; solved: [#98](https://github.com/volodya-lombrozo/jsmith/issues/98).

**Comment by 0pdd** [comment]
@volodya-lombrozo 4 puzzles [#100](https://github.com/volodya-lombrozo/jsmith/issues/100), [#105](https://github.com/volodya-lombrozo/jsmith/issues/105), [#106](https://github.com/volodya-lombrozo/jsmith/issues/106), [#97](https://github.com/volodya-lombrozo/jsmith/issues/97) are still not solved; solved: [#98](https://github.com/volodya-lombrozo/jsmith/issues/98), [#99](https://github.com/volodya-lombrozo/jsmith/issues/99).

**Comment by 0pdd** [comment]
@volodya-lombrozo 3 puzzles [#100](https://github.com/volodya-lombrozo/jsmith/issues/100), [#105](https://github.com/volodya-lombrozo/jsmith/issues/105), [#97](https://github.com/volodya-lombrozo/jsmith/issues/97) are still not solved; solved: [#106](https://github.com/volodya-lombrozo/jsmith/issues/106), [#98](https://github.com/volodya-lombrozo/jsmith/issues/98), [#99](https://github.com/volodya-lombrozo/jsmith/issues/99).

**Comment by 0pdd** [comment]
@volodya-lombrozo 2 puzzles [#100](https://github.com/volodya-lombrozo/jsmith/issues/100), [#97](https://github.com/volodya-lombrozo/jsmith/issues/97) are still not solved; solved: [#105](https://github.com/volodya-lombrozo/jsmith/issues/105), [#106](https://github.com/volodya-lombrozo/jsmith/issues/106), [#98](https://github.com/volodya-lombrozo/jsmith/issues/98), [#99](https://github.com/volodya-lombrozo/jsmith/issues/99).
